Extend "antctl mc get joinconfig" to print member token Secret

jianjuns opened the pull request:
Add a "--member-token TOKEN" option to print the token Secret manifest with the member join parameters.
Also fixed the "get membertoken" YAML output and "join" command code to support creating the token Secret from the "get membertoken" output.

Signed-off-by: Jianjun Shen <shenj@vmware.com>

luolanzone commented on pkg/antctl/raw/multicluster/get/member_token.go:
```diff
@@ -107,41 +103,41 @@ func runEToken(cmd *cobra.Command, args []string) error {
 		}
 
-		memberTokens = append(memberTokens, memberToken)
+		outToken := &memberToken
```
I think we should also ignore the token if it has no `common.CreateByAntctlAnnotation`?

jianjuns replied:
I feel for getting a single token, no need to check the token is created by antctl or not. Why we do not support getting a token created by other ways?

luolanzone commented on pkg/antctl/raw/multicluster/get/member_token.go:
```diff
@@ -107,41 +103,41 @@ func runEToken(cmd *cobra.Command, args []string) error {
-		if singleToken {
-			return fmt.Errorf("Member token %s created by antctl is not found", args[0])
+		if len(opaqueMemberTokens) == 0 {
+			if optionsToken.namespace != "" {
+				fmt.Fprintf(cmd.ErrOrStderr(), "No token found in Namespace %s\n", optionsToken.namespace)
+			} else {
+				fmt.Fprintln(cmd.ErrOrStderr(), "No token found")
+			}
```
namespace check can be removed since you already handled single token case before.
```suggestion
		fmt.Fprintln(cmd.ErrOrStderr(), "No token found")
```

jianjuns replied:
I feel no issue to print the Namespace if Namespace is provided. But sure I can remove it if you prefer.

jianjuns replied:
Just checked other commands and found they all print Namespace when Namespace is provided. I would keep this command the same to avoid more changes.

luolanzone replied:
You are right, I misread the codes.

luolanzone commented on pkg/antctl/raw/multicluster/init.go:
```diff
@@ -112,6 +115,23 @@ func initRunE(cmd *cobra.Command, args []string) error {
+	var tokenSecret *corev1.Secret
+	if initOpts.createToken {
+		if err := common.CreateMemberToken(cmd, initOpts.k8sClient, defaultToken, initOpts.namespace, &createdRes); err != nil {
+			fmt.Fprintf(cmd.ErrOrStderr(), "Failed to create member token\n")
```
Maybe we can add a message here to notify user that they can retry with `antctl mc create membertoken` later?

jianjuns replied:
Sounds like a good idea. Added.

luolanzone reviewed (commented):
There are a few golang-lint issues, could you check and fix it? 
You may also update the doc `docs/multicluster/antctl.md` to reflect the change of `antctl mc get joinconfig`

jianjuns commented on pkg/antctl/raw/multicluster/get/member_token.go:
```diff
@@ -107,41 +103,41 @@ func runEToken(cmd *cobra.Command, args []string) error {
+			outToken = common.ConvertMemberTokenSecret(outToken)
+		}
+		return output(*outToken, true, optionsToken.outputFormat, cmd.OutOrStdout(), membertoken.Transform)
 	} else {
```
I chose to keep the original way. But sure I can remove it if you prefer.

jianjuns commented:
> There are a few golang-lint issues, could you check and fix it? 

Fixed.

> You may also update the doc `docs/multicluster/antctl.md` to reflect the change of `antctl mc get joinconfig`

Updated.

luolanzone reviewed (approved):
LGTM
